src/radon.ts

```diff
@@ -1,423 +1,527 @@
-      scriptId: 0,
-      label,
-      selected: this.wrapResultInCache(this.generateSelectedReducerArgument(reducer)),
-    } as CachedMarkupSelect
+  public getLastOperator(): Operator | null {
+    return this.operators.length ? this.operators[this.operators.length - 1] : null
   }
 
-  public generateSelectedFilterArgument(
-    filterArgument: FilterArgument
-  ): CachedMarkupSelectedOption {
-    const filter: Filter = filterArgument[0]
-    const argument = filterArgument[1]
-    const selectedArgument: CachedMarkupSelectedOption = {
-      arguments: [this.wrapResultInCache(this.generateInputArgument(argument))],
-      label: Filter[filter],
-      hierarchicalType: MarkupHierarchicalType.SelectedOperatorOption,
-      markupType: MarkupType.Option,
-      outputType: OutputType.Bytes,
-    }
-    return selectedArgument
+  public push(operator: MirOperator) {
+    this.operators.push(
+      new Operator(
+        this.cache,
+        this.scriptId,
+        this.getOutputType(),
+        operator,
+        this.onChildrenEvent()
+      )
+    )
   }
 
-  public generateSelectedReducerArgument(reducer: Reducer): MarkupSelectedOption {
-    const selectedArgument: MarkupSelectedOption = {
-      arguments: [],
-      label: Reducer[reducer],
-      hierarchicalType: MarkupHierarchicalType.SelectedOperatorOption,
-      markupType: MarkupType.Option,
-      outputType: OutputType.Bytes,
+  public addOperator() {
+    const lastOutputType = this.getOutputType()
+    const type: Type | null = fromOutputTypeToType(lastOutputType)
+
+    if (type) {
+      const operator: MirOperator = getDefaultMirOperatorByType(type)
+      this.operators.push(
+        new Operator(this.cache, this.scriptId, lastOutputType, operator, this.onChildrenEvent())
+      )
+    } else {
+      // TODO: search in operators the type for the regarding types:
+      // SubscriptOutput, ReducerOutput, FilterOutput, MatchOutput, Same, Inner
+      this.operators.push(
+        new Operator(this.cache, this.scriptId, lastOutputType, null, this.onChildrenEvent())
+      )
     }
-    return selectedArgument
   }
+}
 
-  // TODO: Remove unknown to have a stronger type
-  public unwrapSource(source: CacheRef): MarkupSource {
-    const cachedMarkupSource: CachedMarkupSource = (this.unwrapResultFromCache(
-      source
-    ) as unknown) as CachedMarkupSource
-    const markupSource: MarkupSource = {
-      url: cachedMarkupSource.url,
-      script: this.unwrapScript(cachedMarkupSource.script),
-    }
-
-    return markupSource
+export class Operator {
+  public cache: Cache
+  public operatorInfo: OperatorInfo
+  public code: OperatorCode
+  public mirArguments: MirArgument[] | []
+  public arguments: Array<Argument>
+  public default: Boolean
+  public scriptId: number
+  public inputType: OutputType
+  public id: number
+  public eventEmitter: EventEmitter
+
+  constructor(
+    cache: Cache,
+    scriptId: number,
+    inputType: OutputType | null,
```

Comment:
Does JavaScript have default arguments?

    inputType: OutputType or DEFAULT_INPUT_TYPE

Reply:
It has but I need to know if it is default or not